growattServer/__init__.py

```diff
@@ -515,3 +515,119 @@ def plant_info(self, plant_id):
 
         data = json.loads(response.content.decode('utf-8'))
         return data
+
+    def get_plant_settings(self, plant_id):
+        """
+        Returns a dictionary containing the settings for the specified plant
+
+        Keyword arguments:
+        plant_id -- The id of the plant you want the settings of
+
+        Returns:
+        A python dictionary containing the settings for the specified plant
+        """
+        response = self.session.get(self.get_url('newPlantAPI.do'), params={
+            'op': 'getPlant',
+            'plantId': plant_id
+        })
+        data = json.loads(response.content.decode('utf-8'))
+        return data
+
+    def modify_plant_settings(self, plant_id, changed_settings):
+        """
+        Convenience function to apply updated settings to the plant (looks up the current settings for you)
+        See README for all possible settings options
+
+        Keyword arguments:
+        plant_id -- The id of the plant you want to change the settings for
+        changed_settings -- A python dictionary containing the settings to be changed and their value
+
+        Returns:
+        A response from the server stating whether the configuration was successful or not
+        """
+        current_settings = self.get_plant_settings(plant_id)
+        return self.apply_plant_settings(current_settings, changed_settings)
+
+
+    def apply_plant_settings(self, current_settings, changed_settings):
```

Comment:
I think i'd prefer to stay closer to the naming of the API endpoints.
So in this case something like `update_plant_settings`

And i think we can get away with making the function do more "magic" 😉 
AKA instead of having a seperate function for if you do not pass the current settings it'll just retrieve them by itself

Reply:
I've updated based on your recommendations. 
Names changes and 'helper' functions removed and replaced by "magic" :)